src/hssm/distribution_utils/dist.py

```diff
@@ -243,12 +267,32 @@ def rng_fn(
                 **kwargs,
             )
 
-            output = np.column_stack([sim_out["rts"], sim_out["choices"]])
+            sims_out = np.column_stack([sim_out["rts"], sim_out["choices"]])
 
             if not is_all_scalar:
-                output = output.reshape((*max_shape[:-1], max_shape[-1] * n_samples, 2))
+                sims_out = sims_out.reshape(
+                    (*max_shape[:-1], max_shape[-1] * n_samples, 2)
+                )
 
-            return output
+            if p_outlier is not None:
+                assert cls._lapse is not None
+                out_shape = sims_out.shape[:-1]
+                replace = rng.binomial(n=1, p=p_outlier, size=out_shape).astype(bool)
+                replace = np.stack([replace, replace], axis=-1)
+                n_draws = np.prod(out_shape)
```

Comment:
np.squeeze ?

Reply:
Not sure what you mean here? np.prod returns a scalar after multiplying all array elements. We need to know how many draws we need from a bernulli distribution

Reply:
ok missed that sims_out.shape[:-1] still includes the 'draws' dimension for e.g. posterior predictive.

Reply:
so mistakenly assumed it's a scalar (in a np.ndarray) anyways.